### ISSUE TITLE
[Feature Request] Private Portals

### ISSUE THREAD

**Issue by Morpheus1101** [SpikeHimself/XPortal#67]
Basically have an option so that a portal can be set so only the individual who made the portal can see the connection or mak eit so anyone can see it.

**Comment by SpikeHimself** [comment]
Hi, thanks for your suggestion!

In issue #58 someone asked for "password tags" - would that align with your request too? If so we can close this issue and you can comment there.

**Comment by Morpheus1101** [comment]
Yes and no tbh.  Id still want them to be open an easily to walk through, but on a busy server, with multriple portals the list gets quite long to scroll through, so if portals could be marked as "Open" so everyone can see them or "Private" so only you can see the portal in the list

**Comment by SpikeHimself** [comment]
And by private do you mean only you should be able to see it? Or others too? How do you envision maintaining access control?

Just fishing for ideas to better understand your request.

**Comment by Morpheus1101** [comment]
Basically if the portal is marked Private, it dosent show up in the list of portals for anyone but the maker of the portal itself unless the portal is un-tagged as Private and then shows up for everyone in the list of available portals.

**Comment by SpikeHimself** [comment]
If I understand correctly this is intended to work only when playing alone? I.e. if you play together with someone else, they wouldn't be able to use travel to your "private" portals. Is that a correct interpretation?

**Comment by Morpheus1101** [comment]
Yes, in multiplayer, if your portal is set to Private nobody but the maker/owner would be able to see that portal name, so when the list of portals is shown, the portal marked Private will not show up in the list.

**Comment by Vapok** [comment]
@Morpheus1101  - Realizing that it's been a minute since this was requested, but I have put up a pull request that adds exactly this. Private Portals, plus Portal Networks, so Portals can be assigned to various player lists, if they are not private.

You can find this merge request here: https://github.com/SpikeHimself/XPortal/pull/138
